core/decorator.py

```diff
@@ -21,16 +22,23 @@ def __init__(self, redirect_to):
 class HttpResponseUnsupportedMediaType(HttpResponse):
     status_code = 415
 
+#replace with cache_control in django 1.10+
+def cache_page_function(function, ttl, shared_cache_maxage=None):
+    """Decorate the provided function by adding Cache-Control and Expires headers to responses"""
 
+    @functools.wraps(function)
+    def decorated_function(*args, **kwargs):
+        response = function(*args, **kwargs)
+        cache.patch_response_headers(response, ttl)
+        maxage = ttl if shared_cache_maxage is None else shared_cache_maxage
+        cache.patch_cache_control(response, public=True, s_maxage=maxage)
+        return response
+
+    return decorated_function
 
-def cache_page(ttl):
+def cache_page(ttl, shared_cache_maxage=None):
```

Comment:
I'm not in love with having two functions with similar names to capture whether the decorator is called with arguments or not. I'm wondering whether it would be better to have one function which basically does `if len(args) and callable(args[0]): …` to determine how it's being called so we can have a single function which works either way, or whether we should simply only use the classic decorator form and require changing e.g. `cache_page_function(views.home.home, settings.DEFAULT_TTL_SECONDS)` to `cache_page_function(views.home.home)(settings.DEFAULT_TTL_SECONDS)`